gcloud/storage/blob.py

```diff
@@ -315,27 +340,46 @@ def download_to_file(self, file_obj, client=None):
         # it has all three (http, API_BASE_URL and build_api_url).
         download.initialize_download(request, client._connection.http)
 
-    def download_to_filename(self, filename, client=None):
+    def download_to_filename(self, filename, key=None, client=None):
         """Download the contents of this blob into a named file.
 
         :type filename: string
         :param filename: A filename to be passed to ``open``.
 
+        :type key: str
+        :param key: Optional 32 byte key for customer-supplied encryption.
+
         :type client: :class:`gcloud.storage.client.Client` or ``NoneType``
         :param client: Optional. The client to use.  If not passed, falls back
                        to the ``client`` stored on the blob's bucket.
 
         :raises: :class:`gcloud.exceptions.NotFound`
         """
         with open(filename, 'wb') as file_obj:
-            self.download_to_file(file_obj, client=client)
+            self.download_to_file(file_obj, key=key, client=client)
 
         mtime = time.mktime(self.updated.timetuple())
         os.utime(file_obj.name, (mtime, mtime))
 
-    def download_as_string(self, client=None):
+    def download_as_string(self, key=None, client=None):
         """Download the contents of this blob as a string.
 
+         Downloading a blob that has been `customer-supplied
+         <https://cloud.google.com/storage/docs/encryption#customer-supplied>`_
+         encryption::
+
+            >>> from gcloud import storage
+            >>> from gcloud.storage import Blob
+
+            >>> sc = storage.Client(project='my-project')
+            >>> bucket = sc.get_bucket('my-bucket')
+            >>> key = 'aa426195405adee2c8081bb9e7e74b19'
+            >>> blob = Blob('secure-data', bucket)
+            >>> data = blob.download_as_string(file_obj, key=key)
```

Comment:
This doc is in `download_to_string` and `download_to_file` but not `download_to_filename`. It seems repetitive to have it in 3 places though (or even 2). Why not just 1? Especially since it is a "minor" feature of downloading.

@jgeewax WDYT?